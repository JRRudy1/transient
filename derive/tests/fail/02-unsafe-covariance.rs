//! Ensure fails when the `covariant` attribute is used without `r#unsafe`
use transient_any::TransientAny;


#[derive(Debug, Clone, PartialEq, Eq, TransientAny)]
#[covariant]
struct S<'a, 'b, T> {
    value1: &'a T,
    value2: &'b T,
}

fn main() {
    // this test should fail to compile
}
