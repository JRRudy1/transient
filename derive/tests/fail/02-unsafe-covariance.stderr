error: At most one lifetime parameter is allowed!
 --> tests/fail/02-unsafe-covariance.rs:7:14
  |
7 | struct S<'a, 'b, T> {
  |              ^^
